File: Minimum Depth of Binary Tree/Minimum Depth of Binary Tree.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int minDepth(TreeNode root) {
        if(root == null)
            return 0;
        int i = minDepth(root.left);
        int j = minDepth(root.right);
        if(i == 0 || j == 0)
            return 1+i+j;
        return Math.min(i,j) + 1; 
    }
}

File: Minimum Depth of Binary Tree/Minimum Depth of Binary Tree.py
# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param root, a tree node
    # @return an integer
    def minDepth(self, root):
        if not root:
            return 0
        l=self.minDepth(root.left)
        r=self.minDepth(root.right)
        if l==0 or r==0:
            return 1+r+l
        return min(l, r)+1
